File: aexample/src/main/java/org/judal/examples/java/jdbc/E31_ConvertToXMLJSON.java
package org.judal.examples.java.jdbc;

import org.junit.Test;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.HashMap;
import java.util.Map;

import org.judal.examples.java.model.Course;
import org.judal.storage.query.relational.RelationalQuery;
import org.judal.storage.table.RecordSet;

public class E31_ConvertToXMLJSON {

	@SuppressWarnings("unused")
	@Test
	public void demo() throws Exception {
		
		setUp();

		Course c = new Course();
		
		c.load(1);

		Map<String,String> attribs = new HashMap<String,String>();
		attribs.put("timestamp", new Date().toString());
		
		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-DD");
		
		String courseAsXML = c.toXML("  ", attribs, dateFormat, null, null);

		try (RelationalQuery<Course> qry = new RelationalQuery<>(Course.class)) {
			qry.setResult("*");
			qry.setFilter("1=1");
						
			RecordSet<Course> courses = qry.fetch();
			
			String coursesAsXML = courses.toXML("  ", dateFormat, null, null);
			
			String coursesAsJSON = courses.toJSON();
		}
		
		tearDown();
	}
	
	public static void setUp() throws Exception {
		E10_WriteCSVDataIntoTheDatabase.setUp();
		E10_WriteCSVDataIntoTheDatabase.insertCoursesIntoDatabase();
	}

	public static void tearDown() throws Exception {
		E10_WriteCSVDataIntoTheDatabase.tearDown();
	}	
	
}

File: aexample/src/main/scala/org/judal/examples/scala/jdbc/E31_ConvertToXMLJSON.scala
package org.judal.examples.scala.jdbc

import org.junit.Test

import java.text.SimpleDateFormat
import java.util.Date

import org.judal.storage.EngineFactory
import org.judal.storage.table.RecordSet
import org.judal.storage.table.IndexableView

import org.judal.examples.scala.model.Course

import org.judal.Using._

import scala.collection.JavaConverters._

/**
 * Example of how to print a RecordSet as XML o JSON
 */
class E31_ConvertToXMLJSON {

	@Test
	def demo() : Unit = {
		
		E31_ConvertToXMLJSON.setUp()

		val c = new Course()
		
		c.load(1)

		val attribs = Map("timestamp" -> new Date().toString).asJava		
		val dateFormat = new SimpleDateFormat("yyyy-MM-DD")
		
		val courseAsXML = c.toXML("  ", attribs, dateFormat, null, null)

		var viw : IndexableView = null
		using (viw) {
		  viw = EngineFactory.getDefaultRelationalDataSource.openIndexedView(c)
						
			val courses : RecordSet[Course] = viw.fetch(c.fetchGroup, "id_course", new Integer(1), new Integer(4))
			
			val coursesAsXML = courses.toXML("  ", dateFormat, null, null)

			val coursesAsJSON = courses.toJSON
		}
		
		E31_ConvertToXMLJSON.tearDown()
	}
		
}


object E31_ConvertToXMLJSON {
  
	def setUp() : Unit = {
		E10_WriteCSVDataIntoTheDatabase.setUp()
		E10_WriteCSVDataIntoTheDatabase.insertCoursesIntoDatabase()
	}

	def tearDown() : Unit = {
		E10_WriteCSVDataIntoTheDatabase.tearDown()
	}	

}